src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzBqJobUserStep.java

```diff
@@ -48,8 +48,11 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
       Map<IamRole, String> sourceDatasetPolicyMap =
           workingMap.get(
               SnapshotWorkingMapKeys.SOURCE_DATASET_POLICY_MAP, new TypeReference<>() {});
-      // Allow the custodian to make queries in this project.
-      policyEmails.add(sourceDatasetPolicyMap.get(IamRole.CUSTODIAN));
+      // Allow the dataset stewards and custodians to make queries in the snapshot project.
+      policyEmails.addAll(
+          List.of(
+              sourceDatasetPolicyMap.get(IamRole.CUSTODIAN),
+              sourceDatasetPolicyMap.get(IamRole.STEWARD)));
```

Comment:
I wonder if it's worth trying to generalize this code. This change makes it even move apparent that our code now knows that a dataset has two roles that transfer permissions to child snapshots. and that this knowledge is in many places in the code.

If we had abstracted that notion before as a method like `Dataset.isInheritedRole()` then this fix could've been in a single place without having to change the code. It's unlikely that we'll need to change this again, but maybe it's worth thinking about that now.
```suggestion
      sourceDatasetPolicyMap.entrySet().stream()
          .filter(entry -> isInheritedRole(entry.getKey()))
          .map(Entry::getValue)
          .forEach(policyEmails::add);
```

Reply:
Ok, I made an attempt at this by adding a method in the DatasetService and then an interface step that could be implemented in each of the three steps we had the same logic. 
I struggled to find a spot to put the `isInhertitedRole` since we really needed it in the DatasetService and in the interface step. I landed on a static DatasetService helper method, but open to other options.